x8.md

```diff
@@ -1,34 +1,38 @@
 ---
 sip: 3
-title: Initial Message Payload Specification Specification
+title: Initial Message Payload Specification
 status: Draft
 type: Standard
-author:
+author: Adam Babik <adam@status.im>, Oskar Thorén <oskar@status.im>
 created: 2019-04-22
 updated:
 ---
 
-**TODO: Remove everyting that doesn't touch on payload here**
-
-Using Adam's spec as starting point. Also be mindful of integration with the overview document.
-
-**TODO: Remove non transport privacy layer aspects here**
-
+- [Payload](#payload)
+- [Encoding](#encoding)
+  - [Future encoding format](#future-encoding-format)
+- [Content types](#content-types)
+- [Message types](#message-types)
+- [Clock vs Timestamp and message ordering](#clock-vs-timestamp-and-message-ordering)
+- [Replies](#replies)
+- [Upgradability](#upgradability)
 
-# Payload
+This specification describes how the payload of each message in the Status Protocol looks like. It does not care how the payload is encrypted or exchanged later.
 
 # Payload
 
-Payload is an array of bytes exchanged between peers. In the case of this protocol, the payload is encoded using [transit format](https://github.com/cognitect/transit-format). It is a text-based format so even encoded version is easily readable by humans. For more information about the payload in Status, please see [Initial Message Payload Specification Specification](x8.md).
-
-Payload contains the following fields:
+Payload is a struct (a compound data type) with the following fields:
 1. text `string`
-2. content type `enum { text/plain }` (more in [Content types](#content-types))
-3. message type `enum { public-group-user-message, user-message, group-user-message }` (more in [Message types](#message-types))
+2. content type `enum` (more in [Content types](#content-types))
+3. message type `enum` (more in [Message types](#message-types))
 4. clock `int64`
 5. timestamp `int64`
 6. content `struct { chat-id string, text string }`
 
+# Encoding
+
+The payload is encoded using [Transit format](https://github.com/cognitect/transit-format). It is a text-based format so even encoded version is easily readable by humans.
```

Comment:
Something on JSON

Perhaps something on brief rationale (for @yenda and @cammellos maybe)

Reply:
Why not JSON you mean?
JSON would work just fine, just larger payload, I believe we had it before. I take transit was chosen mainly to reduce bandwitdth, but not sure.

Reply:
I mean it's "similar" but more efficient. Different from saying text based like IRC

Reply:
Reducing bandwidth sounds like a valid reason.

What's wrong with saying "text-based"? Both JSON and Transit are text-based. Unless you mean that it was not the reason of choosing it and this part suggests so.

Reply:
Yes the reason for transit over json was bandwidth and better clojure types supports, because we wanted to have an upgradable protocol at the time (not extensibility) we opted to use records, which save bandwidth as fields are known to both sides and therefore it is sent in an array. The record name was also shorthanded to represent message type and message type version. But as stated above this was shortcuted later on for extensibility and therefore there never used, so the array/record is completely useless and makes extensibility harder without a proper versioning scheme to keep track of changes